google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectRepositoriesModelItem.java

```diff
@@ -53,26 +50,36 @@ public ProjectRepositoriesModelItem(@NotNull String cloudProject,
     setUserObject(cloudProject);
   }
 
-  public void loadRepositories() {
-    try {
-      ListReposResponse response = cloudRepositoryService.list(user, cloudProject);
-
-      removeAllChildren();
-      List<Repo> repositories = response.getRepos();
-      if (!response.isEmpty() && repositories != null) {
-        repositories.forEach(repo -> {
-          Object name = repo.get("name");
-          if (name != null) {
-            add(new RepositoryModelItem(name.toString()));
+  public void loadRepositories(Runnable onComplete) {
+    cloudRepositoryService
+        .listAsync(user, cloudProject)
+        .thenAccept(response -> {
+          if (response == null) {
+            return;
           }
+
+          removeAllChildren();
+          List<Repo> repositories = response.getRepos();
+          if (!response.isEmpty() && repositories != null) {
+            repositories.forEach(repo -> {
+              Object name = repo.get("name");
+              if (name != null) {
+                add(new RepositoryModelItem(name.toString()));
+              }
+            });
+          } else {
+            add(new ResourceEmptyModelItem(GctBundle.message("cloud.repository.list.empty")));
+          }
+
+          onComplete.run();
+        })
+        .exceptionally(response -> {
+          removeAllChildren();
+          add(new ResourceErrorModelItem(GctBundle.message("cloud.repository.list.error")));
+
+          onComplete.run();
+          return null;
```

Comment:
it requires a return value. I couldn't find a cleaner way to handle the exception case. The exceptionally method requires a lambda returning another CompletableFuture since its a fluent API.